tasks/setup-Debian.yml

```diff
@@ -6,6 +6,30 @@
       - docker-engine
     state: absent
 
+- name: Install rootless docker dependencies
+  apt:
+    name:
+      - dbus-user-session
+    state: present
+  when: docker_rootless == true
+
+- name: Install rootless docker dependencies Debian
+  apt:
+    name:
+      - slirp4netns
+    state: present
+  when: docker_rootless == true and ansible_distribution == 'Debian'
+
+- name: Add kernel.unprivileged_userns_clone=1 to /etc/sysctl.conf
+  lineinfile:
+    path: /etc/sysctl.conf
+    insertafter: EOF
+    line: 'kernel.unprivileged_userns_clone=1'
+  when:
+    - docker_rootless == true
+    - ansible_distribution == 'Debian'
+    - ansible_distribution_version == '10'
```

Comment:
Would this also apply to later versions of Debian, or *only* 10?

Reply:
This only applies to Debian 10.  The [docs](https://docs.docker.com/engine/security/rootless/) say:

> For Debian 10, add kernel.unprivileged_userns_clone=1 to /etc/sysctl.conf (or /etc/sysctl.d) and run sudo sysctl --system. This step is not required on Debian 11.

Although, technically I haven't tested on a pure Debian machine yet, just Ubuntu.  I'll try to test it today but I hope I don't hit another subscription wall like I did with RHEL 😅

Reply:
Ok so I tested this on Debian 10 and it looks like the docs are a bit misleading.  It says `slirp4netns >= 0.4.0` is required, however, that version doesn't seem to be available using the package manager.

I tested it on Debian 11 though and it works just fine, so I've removed the extra setup needed for Debian 10 altogether.

So all-in-all, it supports all of the latest versions, Ubuntu 20 and above, CentOS Stream 9, and Debian 11.

I could update the docs and state that this role only supports rootless docker for the latest versions, and maybe put a disclaimer that says, "rootless docker is still experimental, and if you need to revert back to a normal installation, you can just run the role again with `docker_rootless` set to false".

I have tested toggling between the two pretty thoroughly and haven't encountered any errors.

Let me know what you think!  Thanks :)